jvm/src/test/scala/org/atnos/site/TransformStack.scala

```diff
@@ -164,19 +166,22 @@ def runAuth[R, U, A](e: Eff[R, A])(implicit
        ax match {
          case Authenticate(token) =>
            // send the TimedFuture effect in the stack U
-           send(authenticate(token)).
+           fromFuture(authenticateImpl(token)).
            // send the Either value in the stack U
            collapse
        }
     })
 
 // call to a service to authenticate tokens
-def authenticate(token: String): TimedFuture[AuthError Either AccessRights] = ???
+def authenticateImpl(token: String): Future[AuthError Either AccessRights] =
+  Future.successful[AuthError Either AccessRights] { Left(AuthError("token invalid!")) }
+
+def authenticate[S: _authenticate](token: String) = Authenticate(token).send
```

Comment:
Just one thing about formatting. Everywhere in the code I have written:
```scala
def authenticate[R :_authenticate]
```
The idea is to be close to a "regular" type bound but with a `_`.